libparsec/crates/client/src/workspace/store/prevent_sync_pattern.rs

```diff
@@ -0,0 +1,117 @@
+use std::sync::Arc;
+
+use libparsec_platform_storage::workspace::{UpdateManifestData, WorkspaceStorage};
+use libparsec_types::{
+    prelude::{anyhow, thiserror},
+    DataError, LocalDevice, LocalFolderManifest, LocalWorkspaceManifest, SecretKey,
+};
+
+#[derive(Debug, thiserror::Error)]
+pub enum ApplyPreventSyncPatternError {
+    #[error("Decryption error")]
+    DecryptionError,
+    #[error(transparent)]
+    Internal(anyhow::Error),
+}
+
+pub(super) async fn ensure_prevent_sync_pattern_applied_to_wksp(
+    storage: &mut WorkspaceStorage,
+    device: Arc<LocalDevice>,
+) -> Result<(), ApplyPreventSyncPatternError> {
+    const PAGE_SIZE: u32 = 1000;
+
+    let (pattern, fully_applied) = storage
+        .get_prevent_sync_pattern()
+        .await
+        .map_err(ApplyPreventSyncPatternError::Internal)?;
+
+    if fully_applied {
+        return Ok(());
+    }
+
+    let mut offset = 0;
+    loop {
+        let manifests = storage
+            .list_manifests(offset, PAGE_SIZE)
+            .await
+            .map_err(ApplyPreventSyncPatternError::Internal)?;
+
+        let mut updated_manifest = Vec::with_capacity(manifests.len());
+        for encoded_manifest in &manifests {
+            if let Some(folder) = decode_folder(encoded_manifest, &device.local_symkey)? {
+                let new_folder =
+                    folder.apply_prevent_sync_pattern(Some(&pattern), device.time_provider.now());
+                // TODO: Is it ok to just check the updated field (DateTime) to see if the manifest has changed,
+                // or better to do a full eq ?
+                if new_folder.updated != folder.updated {
```

Comment:
?

Reply:
For now you should go with a full eq (this way play safe and avoid weird bugs !)

Then, you should open an issue to indicate we should modify `apply_prevent_sync_pattern` to make it returns a `Option<Self>`  so that we know if the manifest has been actually modified

Reply:
done